code/game/objects/items.dm

```diff
@@ -100,6 +100,8 @@ GLOBAL_DATUM_INIT(fire_overlay, /image, image("icon" = 'icons/goonstation/effect
 	var/tool_enabled = TRUE //If we can turn on or off, are we currently active? Mostly for welders and this will normally be TRUE
 	var/tool_volume = 50 //How loud are we when we use our tool?
 	var/toolspeed = 1 // If this item is a tool, the speed multiplier
+	/// Show what a machine/door wire does when held.
+	var/show_wire_information = FALSE
```

Comment:
Not sure this should be put on base level `/obj/item`

Reply:
Yeah I wasn't sure about that bit either. I can't see another way of doing it though, since there's not an `/obj/item/tool` parent that it could be put on. (There probably should be though)
_Maybe_ a component or something would work better, but I've never actually looked into how those work so I don't know.

Reply:
Suppose you could make it proc-based instead. For example:
```dm
/obj/item/proc/shows_wire_info()
    return FALSE

/obj/item/multitool/cyborg/drone/shows_wire_info()
    return TRUE

/obj/item/wirecutters/cyborg/drone/shows_wire_info()
    return TRUE

// etc.
```
If that's not wanted, then yeah could go component (or element) or maybe even pure signal based.

Reply:
Not sure which of these would be preferred, but I'll look into them. All of the actual wire code is there already so it should be pretty easy.

Reply:
Might have forgotten about this until just now, whoops.
So what *would* be the best way to do this? Changing it to use procs instead might be better, but it also has the same issue of being something on `/obj/item`, and only used for a couple of subtypes.

Reply:
Besides the things I listed before, you could make a trait `SHOWS_WIRE_INFO` or such, and attach it to the items you want. Then check for the trait in the wires datum. This is probably a better way than the others.